WeakAurasOptions/AuthorOptions.lua

```diff
@@ -47,98 +48,87 @@
       bigStep (optional) -> step size of the slider. Defaults to 0.05
       step (optional) -> like bigStep, but applies to number input as well
 ]]
-
 local WeakAuras = WeakAuras
 local L = WeakAuras.L
 
 local tinsert, tremove, tconcat = table.insert, table.remove, table.concat
 local conflictBlue = "|cFF4080FF"
-local references, conflict = {}, {} -- magic values
--- classifications of options. Currently, we have simple and noninteractive
-local optionClasses = {
-  toggle = "simple",
-  input = "simple",
-  number = "simple",
-  range = "simple",
-  color = "simple",
-  select = "simple",
-  multiselect = "simple",
-  description = "noninteractive",
-  space = "noninteractive",
-  header = "noninteractive",
-}
+local conflict = {} -- magic value
 
-local function atLeastOneSet(data, references, key)
-  for childID, optionID in pairs(references) do
-    if data[childID].authorOptions[optionID][key] ~= nil then
+local optionClasses = WeakAuras.author_option_classes
+
+local function atLeastOneSet(references, key)
```

Comment:
A very large percentage of the changes in this file are due to a simplification which I only thought of recently. Essentially, a single selected aura is now considered to be a group with just one member. This simplified the reasoning process greatly, and should reduce the number of stupid bugs due to neglecting to change the multi selected code path or whatever.